runtime/src/chplsys.c

```diff
@@ -502,19 +502,39 @@ uint64_t chpl_sys_availMemoryBytes(void) {
 //
 // Return information about the processors on the system.
 //
-static void getCpuInfo(int* p_numPhysCpus, int* p_numLogCpus) {
-  //
-  // Currently this is pretty limited -- it only returns the number of
-  // physical and logical (hyperthread, e.g.) CPUs, mostly relies on
-  // /proc/cpuinfo, and only supports homogeneous compute nodes with
-  // the same number of cores and siblings on every physical CPU.  It
-  // will probably need to become more complicated in the future.
-  //
+static struct {
+  int physId, coreId;
+} cpuTab[1024];
```

Comment:
Does this mean we'll fail on machines that have more than 1024 cores? That seems like an unfortunate limitation. I think the largest the kernel currently supports is 8192: https://github.com/torvalds/linux/blob/1bad9ce155a7c010a9a5f3261ad12a6a8eccfb2c/arch/x86/Kconfig#L991-L1002

I'd prefer to go with 8192 here, though that'll only become obsolete over time, so I wonder if there's a more programatic way to check.

Reply:
I could count the newlines in `/proc/cpuinfo` first and dyamically allocate the table to that size, then rewind the file and process it for real.  Surely the system cannot have more CPUs, either physical or logical, than the number of lines in `/proc/cpuinfo`.

More seriously, I can certainly make the table dynamically sized.